### DISCUSSION TITLE
Vitest is hanging on setTimeouts and promises

### DISCUSSION THREAD

**Discussion by ilearnio** [vitest-dev/vitest#3819] [Ideas]
I really like vitest in general after being using if for the past week or so, but these existing issues are so annoying that I'm even thinking to switch to plain old Mocha. Here's why:

- It does some magic to the console logs which I don't like, apparently because of it's complex multy-threading design. I don't like it if I put a console log somewhere, sometimes it doesn't display at all until the test process stops executing. This is especially annoying for a case when debugging an issue that when vitest process hangs, or my test execute for too long, so this becomes a real pain in ass. If I added a console log, I expect it to be shown no matter what, without complex magic around the console happening behind the scenes.

- It hangs on timeouts, it hangs on promises and there is no way to debug the issue. Setting global timeout doesn't have impact, it still just hangs indefinitely. And, sometimes (rarely but still inredibly annoying) pressing Ctrl+C to exit the process doesn't help there is no way to stop the execution ecept for force killing the process. Again, the output of `console.log`s is swallowed too so I cannot even use the poking and prodding method to debug my tests. So I don't know where it hangs, why it hangs, vitest itself shows no debugging info, and there is no way to debug it with console logs, and I cannot even stop the process to rerun the tests.

Sorry if I'm overemotional, just spent an hour already on trying to find where the issue is but with no luck still. 

PS: vitest/0.32.4 darwin-arm64 node-v16.17.0

**Comment by sheremet-va** [comment]
I agree that console log magic is annoying.

Not sure what you mean by timeout hangouts though.
